Add tiff images support

EkaterinaPoddubskaya opened the pull request:
This PR resolves issue #123

EkaterinaPoddubskaya committed:
Add ol library

EkaterinaPoddubskaya committed:
Add div for zooming image to templates, delete unused code

EkaterinaPoddubskaya committed:
Implement zoom for tif, jpg, png formats

EkaterinaPoddubskaya committed:
Remove unused dependency

EkaterinaPoddubskaya committed:
Fix typos

EkaterinaPoddubskaya committed:
Remove dead and duplicate code

EkaterinaPoddubskaya committed:
Fix tree checkbox collapsing behavior

EkaterinaPoddubskaya committed:
Add viewport meta tag to HTML pages

EkaterinaPoddubskaya committed:
Add image zoom state synchronization for metadata toggle

EkaterinaPoddubskaya committed:
Reset image view extent on container resize

EkaterinaPoddubskaya committed:
Fix 404 error on redirect with special character filters (/^:<>)

EkaterinaPoddubskaya committed:
Rename image enlargement window

EkaterinaPoddubskaya committed:
Adapt window sizes to screen resolution, delete unused variable

EkaterinaPoddubskaya committed:
Increase anchor icon width

EkaterinaPoddubskaya committed:
Change mobile header layout to show filtered images count

EkaterinaPoddubskaya committed:
Refactor anchor icon and group dropdown creation to avoid duplication

EkaterinaPoddubskaya committed:
Make adaptive design for image name, checkbox, icons, and icon badges

EkaterinaPoddubskaya commented:
Hi @danlamanna,
Could you please re-enable the ability to log in via this link: https://add-tiff-support.isic-gallery.pages.dev/? I would greatly appreciate it

EkaterinaPoddubskaya committed:
Encapsulate image element resizing and image view extent restoration logic

EkaterinaPoddubskaya committed:
Delete whitespace

danlamanna commented:
Sorry, we didn't disable it intentionally we just broke something on our end. It should be fixed (and stay that way 🤞) now.

EkaterinaPoddubskaya committed:
Adjust image view center for smoother zoom

EkaterinaPoddubskaya committed:
Align zoom buttons and metadata toggle

EkaterinaPoddubskaya committed:
Adapt font size to fit inside container

EkaterinaPoddubskaya committed:
Hide filtered download button when searching by name

EkaterinaPoddubskaya committed:
Fix three state checkbox behavior

Checkbox supports three states: checked, unchecked, and partially checked.
Previously, partially checked meant that not all matching images were displayed.
Now, it reflects that the user has selected not all child options, independent of the number of displayed images.
Added hasHiddenOption flag to indicate if the current option has images not associated with any of its children.

EkaterinaPoddubskaya committed:
Fix suggest highlighting parent and children options

EkaterinaPoddubskaya committed:
Fix unchecking first parent in grandparent after selecting all children

EkaterinaPoddubskaya committed:
Align multi-image lesion window header elements

EkaterinaPoddubskaya committed:
Fix selecting affected items in onShow event

EkaterinaPoddubskaya committed:
Fix eslint errors

EkaterinaPoddubskaya committed:
Fix suggest highlight for clicked option parents

brianhelba commented:
@neuroelf Can you review this and let us know if you accept?

If so, let's merge it.
